CMake/FindArmadillo.cmake

```diff
@@ -65,305 +70,63 @@ if(ARMADILLO_INCLUDE_DIR)
-      set(SUPPORT_LIBRARIES "${HDF5_LIBRARIES}")
-    endif()
-
-    # Versions between 4.300 and 4.500 did successfully wrap HDF5, but didn't have good support for setting the include directory correctly.
-    if(NOT "${ARMA_USE_HDF5}" STREQUAL "" AND
-       "${ARMADILLO_VERSION_STRING}" VERSION_GREATER "4.299.0" AND
-       "${ARMADILLO_VERSION_STRING}" VERSION_LESS "4.450.0")
-      message(STATUS "Armadillo HDF5 support is enabled and include "
-                     "directories must be found.")
-      find_package(HDF5)
-
-      if(NOT HDF5_FOUND)
-        # On Debian systems, the HDF5 package has been split into multiple
-        # packages so that it is co-installable.  But this may mean that the
-        # include files are hidden somewhere very odd that the FindHDF5.cmake
-        # script will not find.  Thus, we'll also quickly check pkgconfig to see
-        # if there is information on what to use there.
-        find_package(PkgConfig)
-        if (PKG_CONFIG_FOUND)
-          pkg_check_modules(HDF5 hdf5)
-        endif()
-      endif()
-
-      if(NOT HDF5_FOUND)
-        # We tried but didn't find it.
-        message(FATAL_ERROR "Armadillo HDF5 support is enabled, but HDF5 "
-            "cannot be found on the system.  Consider disabling HDF5 support.")
-      endif()
-
-      set(SUPPORT_INCLUDE_DIRS "${HDF5_INCLUDE_DIRS}")
-    endif()
+include(FindPackageHandleStandardArgs)
 
-  endif()
-else()
-  message(FATAL_ERROR "${ARMADILLO_INCLUDE_DIR}/armadillo_bits/config.hpp not "
-      "found!  Cannot determine what to link against.")
-endif()
+# if ARMA_USE_WRAPPER is set, then we just link to armadillo, but if it's not then we need support libraries instead
+set(ARMA_SUPPORT_LIBRARIES)
 
-if (ARMA_NEED_LIBRARY)
+if(ARMA_USE_WRAPPER)
   # UNIX paths are standard, no need to write.
   find_library(ARMADILLO_LIBRARY
     NAMES armadillo
     PATHS "$ENV{ProgramFiles}/Armadillo/lib"  "$ENV{ProgramFiles}/Armadillo/lib64" "$ENV{ProgramFiles}/Armadillo"
     )
-
-  # Checks 'REQUIRED', 'QUIET' and versions.
-  include(FindPackageHandleStandardArgs)
   find_package_handle_standard_args(Armadillo
     REQUIRED_VARS ARMADILLO_LIBRARY ARMADILLO_INCLUDE_DIR
     VERSION_VAR ARMADILLO_VERSION_STRING)
-  # version_var fails with cmake < 2.8.4.
-else ()
-  # Checks 'REQUIRED', 'QUIET' and versions.
-  include(FindPackageHandleStandardArgs)
+else(ARMA_USE_WRAPPER)
+  # don't link to armadillo in this case
+  set(ARMADILLO_LIBRARY "")
+  if(ARMA_USE_LAPACK)
+    find_package(LAPACK REQUIRED)
+    set(ARMA_SUPPORT_LIBRARIES "${ARMA_SUPPORT_LIBRARIES}" "${LAPACK_LIBRARIES}")
+  endif(ARMA_USE_LAPACK)
+  if(ARMA_USE_BLAS)
+    find_package(BLAS REQUIRED)
+    set(ARMA_SUPPORT_LIBRARIES "${ARMA_SUPPORT_LIBRARIES}" "${BLAS_LIBRARIES}")
+  endif(ARMA_USE_BLAS)
+  if(ARMA_USE_ARPACK)
+    find_package(ARPACK REQUIRED)
+    set(ARMA_SUPPORT_LIBRARIES "${ARMA_SUPPORT_LIBRARIES}" "${ARPACK_LIBRARIES}")
+  endif(ARMA_USE_ARPACK)
```

Comment:
I think we'll need to handle `ARMA_USE_HDF5` too.  I do remember a *lot* of fighting with that when I initially wrote this script, so maybe some of that code should be reused instead of just a simple call to `find_package(HDF5)`.  Anyway, maybe some of it's out of date now; I'm not sure how much the HDF5 packaging situation has changed.

Reply:
Ah, I had misread the HDF5 section in the original findArmadillo and thought it only applied to Armadillo < 4.300